### ISSUE TITLE
It seems impossible to search in casks description 

### ISSUE THREAD

**Issue by fad1** [Homebrew/homebrew-cask#108030]
### Provide a detailed description of the proposed feature

It should be possible to search in casks description

### What is the motivation for the feature?

Ability to find the best casks that do something.

### Example use case

I'm not sure but it seems impossible to search casks' descriptions.

As an experiment, I'm looking for all casks that have 'markdown' in their descriptions.

I tried many commands like 

> brew search --cask --desc markdown

The cask 'macdown' is not returned, no matter what I try. 
https://formulae.brew.sh/cask/macdown

Is it actually impossible to search in casks descriptions?

**Comment by bevanjkay** [comment]
If you run `brew search --help` it says that only formula can be search by description.

> --desc                       Search for formulae with a description
>                                    matching text and casks with a name
>                                    matching text.

**Comment by fad1** [comment]
Makes sense, I edited the title of the request, thanks!

**Comment by Rylan12** [comment]
I agree this is something we should have. I'm moving this to Homebrew/brew as that's where the change should be made.

At a minimum, if `--cask` and `--desc` are mutually exclusive, a conflict should be declared in `brew search`
